packages/engine/src/interaction/components/properties/InteractableComponentNodeEditor.tsx

```diff
@@ -0,0 +1,180 @@
+import StringInput from '@etherealengine/editor/src/components/inputs/StringInput'
+import NodeEditor from '@etherealengine/editor/src/components/properties/NodeEditor'
+import {
+  EditorComponentType,
+  commitProperties,
+  commitProperty,
+  updateProperty
+} from '@etherealengine/editor/src/components/properties/Util'
+import { InteractableComponent } from '../InteractableComponent'
+
+type OptionsType = Array<{
+  callbacks: Array<{
+    label: string
+    value: string
+  }>
+  label: string
+  value: string
+}>
+
+const callbackQuery = defineQuery([CallbackComponent])
+
+export const InteractableComponentNodeEditor: EditorComponentType = (props) => {
+  const { t } = useTranslation()
+  const targets = useState<OptionsType>([{ label: 'Self', value: 'Self', callbacks: [] }])
+
+  const interactableComponent = useComponent(props.entity, InteractableComponent)
+
+  useEffect(() => {
+    const options = [] as OptionsType
+    options.push({
+      label: 'Self',
+      value: 'Self',
+      callbacks: []
+    })
+    for (const entity of callbackQuery()) {
+      if (entity === props.entity || !hasComponent(entity, EntityTreeComponent)) continue
+      const callbacks = getComponent(entity, CallbackComponent)
+      options.push({
+        label: getComponent(entity, NameComponent),
+        value: getComponent(entity, UUIDComponent),
+        callbacks: Object.keys(callbacks).map((cb) => {
+          return { label: cb, value: cb }
+        })
+      })
+    }
+    targets.set(options)
+  }, [])
+
+  const addCallback = () => {
+    const label = ''
+    const callbacks = [
+      ...interactableComponent.callbacks.value,
+      {
+        target: 'Self',
+        callbackID: ''
+      }
+    ]
+    commitProperties(InteractableComponent, { label: label, callbacks: JSON.parse(JSON.stringify(callbacks)) }, [
+      props.entity
+    ])
+  }
+  const removeCallback = (index: number) => {
+    const callbacks = [...interactableComponent.callbacks.value]
+    callbacks.splice(index, 1)
+    commitProperties(InteractableComponent, { callbacks: JSON.parse(JSON.stringify(callbacks)) }, [props.entity])
+  }
+
+  return (
+    <NodeEditor
+      {...props}
+      name={t('editor:properties.interactable.name')}
+      description={t('editor:properties.interactable.description')}
+    >
+      <InputGroup name="Label" label={t('editor:properties.interactable.lbl-label')}>
+        <StringInput
+          value={interactableComponent.label.value!}
+          onChange={updateProperty(InteractableComponent, 'label')}
+          onRelease={commitProperty(InteractableComponent, 'label')}
+        />
+      </InputGroup>
+      <PropertiesPanelButton type="submit" onClick={addCallback}>
+        {t('editor:properties.interactable.addcallbacktitle')}
+      </PropertiesPanelButton>
+
+      <div key={`callback-list-${props.entity}`}>
+        {interactableComponent.callbacks.map((callback, index) => {
```

Comment:
Need to key with the index the return value

Reply:
addressed